### ISSUE TITLE
Command "find" in env.bat might not work as expected

### ISSUE THREAD

**Issue by rstahn** [winpython/winpython#421]
I noticed a long startup-time of the WinPython command prompt since Release 2016-03.
The command window opens but shows strange error messages:
```
find: /C/Config.Msi: Permission denied
find: /C/hiberfil.sys: Permission denied
find: /C/MSOCache: Permission denied
find: /C/pagefile.sys: Permission denied
find: /I: No such file or directory
find: ;C:\FWTools\WinPython-201603\scripts\..\python-3.4.4\;: No such file or directory
```
Only after minutes of heavy HDD activity the command prompt finally shows up.

Investigation of this behavior leads to the following code in `env.bat` added with Issue #310:
```
echo ;%PATH%; | find /C /I ";%WINPYDIR%\;" >nul
if %ERRORLEVEL% NEQ 0 set PATH=%WINPYDIR%\Lib\site-packages\PyQt5;%WINPYDIR%\Lib\site-packages\PyQt4;%WINPYDIR%\;%WINPYDIR%\DLLs;%WINPYDIR%\Scripts;%WINPYDIR%\..\tools;%WINPYDIR%\..\tools\mingw32\bin;%WINPYDIR%\..\tools\R\bin\i386;%WINPYDIR%\..\tools\Julia\bin;%PATH%;
```

This code works fine as long as the command `find` which ships with Windows is used.
In my case it was shadowed by an executable `find.exe` installed by a [WinAVR ](http://winavr.sourceforge.net) distribution, which has a different behavior (actually searching the current drive completely).

Though this is a rather specific problem I like to suggest to improve the above code, which tries to determine whether the above WinPython directories are already listed in the PATH environment variable. 

Maybe the "search a string for a substring" problem can be solved without running an external 'find' command?

A possible solution is proposed by this [Stackoverflow answer](http://stackoverflow.com/questions/7005951/batch-file-find-if-substring-is-in-string-not-in-a-file), although I do not know if this solution can handle white spaces within the PATH string.

**Comment by stonebig** [comment]
best suggestion: rename the "find.exe" of WinAVR to "findAVR.exe"

**Comment by rstahn** [comment]
Well - I agree that renaming the offending file is the obvious workaround. But this does not solve the problem for users who need to work with tools like WinAVR (though it may be an incorrect practice to replace system commands) and who are not even aware of the root cause of the unusual behavior of WinPython. It actually took me some time to track this down. 

Furthermore, its not only the rather exotic WinAVR that installs an incompatible replacement for `find` but also the much more widespread Cygwin. Cygwin even [recommends on its website](https://www.cygwin.com/cygwin-ug-net/using-effectively.html) to modify `PATH` such that the Cygwin version of `find` is found first:
> A few Windows tools, such as find.exe, link.exe and sort.exe, may conflict with the Cygwin versions make sure that you use the full path (/usr/bin/find) or that your Cygwin bin directory comes first in your PATH.

I had another look for possible solutions and found a very pragmatic approach [here](http://stackoverflow.com/a/22332510): 
Simply make sure the correct version of `find` is used by explicitly specifying the path to the windows version: `%SYSTEMROOT%\system32\find.exe`. The code in `env.bat` should then look like this:
```
echo ;%PATH%; | %SYSTEMROOT%\system32\find.exe /C /I ";%WINPYDIR%\;" >nul
```
This is a low-risk solution that should work on all versions of Windows. What do you think?

**Comment by stonebig** [comment]
oups ! I was reading at your provided code and forgot to check on current one... sorry.

This change has been made september 5th:

https://github.com/winpython/winpython/commit/62401d674acca39a23dc8742366ef344fd6d8bbc

````
set FINDDIR=%WINDIR%\system32
echo ;%PATH%; | %FINDDIR%\find.exe /C /I ";%WINPYDIR%\;" >nul
````

**Comment by stonebig** [comment]
I don't see the subtle nuance between %SYSTEMROOT% and %WINDIR%, though.

**Comment by rstahn** [comment]
Oh, I did not check the current code either - only the latest 2016-04 release.
So this is actually already fixed. Great!

Regarding the difference between %SYSTEMROOT% and %WINDIR% [this site](http://ss64.com/nt/syntax-variables.html) writes:
> %WinDir% pre-dates Windows NT and seems to be superseded by %SystemRoot%
Set by default as windir=%SystemRoot%
%windir% is a regular variable and can be changed, which makes it less robust than %systemroot% 

So %SYSTEMROOT% is read-only and cannot be modified by other third-party software. So using it would be even more save than %WINDIR%.
